### ISSUE TITLE
Don't use arguments[0]

### ISSUE THREAD

**Issue by FokkeZB** [appcelerator/kitchensink-v2#1]
Although I welcome the use of a SEF to group code that runs on creation, I don't think it should use `arguments[0]` as the args are already available via `$.args` (see [component.js](https://github.com/appcelerator/alloy/blob/master/Alloy/template/component.js#L20)).

So instead of:

```js
/**
 * The scoped constructor of the controller.
 * @param args {Object} The arguments passed to initialize the controller.
 **/
(function constructor(args) {
    
})(arguments[0] || {});
```

Do:

```js
/**
 * The scoped constructor of the controller.
 **/
(function constructor() {
    
})();
```

Or if you want to point out the existence of `$.args` use the 2 lines the [generator adds](https://github.com/appcelerator/alloy/blob/master/Alloy/template/controller.js#L1-L2):

```js
// Arguments passed into this controller can be accessed via the `$.args` object directly or:
var args = $.args;

/**
 * The scoped constructor of the controller.
 **/
(function constructor() {
    
})();
```

Or if you must:

```js
/**
 * The scoped constructor of the controller.
 * @param args {Object} The arguments passed to initialize the controller.
 **/
(function constructor(args) {
    
})($.args);
```

**Comment by hansemannn** [comment]
Thanks Fokke! Will migrate in the next patch version.

**Comment by hansemannn** [comment]
Fixed in @dd3bdcd 🚀
